internal/wasm/sdk/opa/nop.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2021 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+// +build !opa_wasm
+
+package opa
+
+import (
+	"context"
+
+	"github.com/open-policy-agent/opa/metrics"
+)
+
+// OPA is a stub implementation of a opa.OPA.
+type OPA struct {
+}
+
+// Result is a stub implementation of a opa.Result.
+type Result struct {
+	Result []byte
+}
+
+// EvalOpts is a stub implementation of a opa.EvalOpts.
+type EvalOpts struct {
+	Entrypoint int32
+	Input      *interface{}
+	Metrics    metrics.Metrics
+}
+
+// New unimplemented.
+func New() *OPA {
+	panic("WebAssembly runtime not supported in this build")
+}
+
+// Init unimplemented.
+func (o *OPA) Init() (*OPA, error) {
+	panic("unreachable")
+}
+
+// SetData unimplemented.
+func (o *OPA) SetData(v interface{}) error {
+	panic("unreachable")
+}
+
+// SetDataPath unimplemented.
+func (o *OPA) SetDataPath(path []string, value interface{}) error {
+	panic("unreachable")
+}
+
+// RemoveDataPath unimplemented.
+func (o *OPA) RemoveDataPath(path []string) error {
+	panic("unreachable")
+}
+
+// SetPolicy unimplemented.
+func (o *OPA) SetPolicy(p []byte) error {
+	panic("unreachable")
+}
+
+// SetPolicyData unimplemented.
+func (o *OPA) SetPolicyData(policy []byte, data *interface{}) error {
+	panic("unreachable")
+}
+
+// Eval unimplemented.
+func (o *OPA) Eval(ctx context.Context, opts EvalOpts) (*Result, error) {
+	panic("unreachable")
+}
+
+// Close unimplemented.
+func (o *OPA) Close() {
+	panic("unreachable")
+}
+
+// Entrypoints unimplemented.
+func (o *OPA) Entrypoints(ctx context.Context) (map[string]int32, error) {
+	panic("unreachable")
+}
+
+// WithPolicyFile unimplemented.
+func (o *OPA) WithPolicyFile(fileName string) *OPA {
+	panic("unreachable")
+}
+
+// WithPolicyBytes unimplemented.
+func (o *OPA) WithPolicyBytes(policy []byte) *OPA {
+	panic("unreachable")
+}
+
+// WithDataFile unimplemented.
+func (o *OPA) WithDataFile(fileName string) *OPA {
+	panic("unreachable")
+}
+
+// WithDataBytes unimplemented.
+func (o *OPA) WithDataBytes(data []byte) *OPA {
+	panic("unreachable")
+}
+
+// WithDataJSON unimplemented.
+func (o *OPA) WithDataJSON(data interface{}) *OPA {
+	panic("unreachable")
+}
+
+// WithMemoryLimits unimplemented.
+func (o *OPA) WithMemoryLimits(min, max uint32) *OPA {
+	panic("unreachable")
+}
+
+// WithPoolSize unimplemented.
+func (o *OPA) WithPoolSize(size uint32) *OPA {
+	panic("unreachable")
+}
+
+// WithErrorLogger unimplemented.
+func (o *OPA) WithErrorLogger(logger func(error)) *OPA {
+	panic("unreachable")
+}
```

Comment:
I think this is OK for now, but I could also imagine this being a little tedious to maintain. That said, I don't know of any better solution off-hand, short of some code generation that surely isn't needed just yet.

Reply:
Done